elasticsearch/src/main/scala/akka/stream/alpakka/elasticsearch/ElasticsearchFlowStage.scala

```diff
@@ -23,33 +23,58 @@ import akka.stream.alpakka.elasticsearch.scaladsl.ElasticsearchSinkSettings
 import org.apache.http.message.BasicHeader
 import org.apache.http.util.EntityUtils
 
-final case class IncomingMessage[T](id: Option[String], source: T)
+object IncomingMessage {
+  // Apply method to use when not using cargo
+  def apply[T](id: Option[String], source: T): IncomingMessage[T, Any] =
+    IncomingMessage(id, source, null)
+
+  // Java-api - without cargo
+  def create[T](id: String, source: T): IncomingMessage[T, Any] =
+    IncomingMessage(Option(id), source)
+
+  // Java-api - with cargo
+  def create[T, C](id: String, source: T, cargo: C): IncomingMessage[T, C] =
+    IncomingMessage(Option(id), source, cargo)
+}
+
+final case class IncomingMessage[T, C](id: Option[String], source: T, cargo: C)
+
+object IncomingMessageResult {
+  // Apply method to use when not using cargo
+  def apply[T](source: T, success: Boolean): IncomingMessageResult[T, Any] =
```

Comment:
It should be used `NotUsed` instead of `Any`.

Reply:
done